### PR TITLE
Bug Fix: Change dropdown role to combobox from menu to fix narrator reading the title twice

### LINKED ISSUE #3319: [Accessibility]: Dropdown is now being treating as menu, thus failing aria-required-children rule
[This ](https://github.com/OfficeDev/office-ui-fabric-react/commit/d57ca247d7333fa86473482e2037e1a5aca2c2f2) commit changed dropdown role from `combobox ` to `menu`.
But it fails https://aka.ms/keros/rules/axe.aria-required-children rule.
Because it has `menu` but no `menuitem` since we render the options with role `listbox` which has valid realtion ship with `combobox` but not a `menu`.

Commit says 

> Change combobox to menu as the dropdown is not editable but the screen reader keeps saying it is

I believe it can addressed with: http://www.w3.org/TR/wai-aria/states_and_properties#aria-autocomplete 

Can we revist and fix this please?

image

https://chrome.google.com/webstore/detail/keros-for-web-accessibili/pbjjkligggfmakdaogkfomddhfmpjeni

Run "fast pass"

### PR DESCRIPTION
**Pull request by sugandhi-ms**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3319 
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Dropdown role was changes from "menu" to "combobox" in the past. This results in narrator to announce the dropdown title, role and selected value twice. 
Changing the role back to "combobox". Also adding aria-readonly='true' since dropdown is not editable. (Narrator announces it as "combobox editable" otherwise)

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
I've got a feeling that it was changed to combobox for a reason though. @betrue-final-final might have some background.

**Comment by betrue-final-final** [comment]
I'm not sure why, and I don't have that deep of knowledge into the engineering aspects of ARIA. I do know that if we announce a dropdown, which is a menu, as a combobox, that will set the wrong expectation.

**Comment by betrue-final-final** [comment]
Ahh, I see. So setting it as read-only. Hmmm, that could work. I'll have to check out the current dropdown and then this PR.

**Comment by sugandhi-ms** [comment]
Correct. The initial change  ( from combobox to menu) was done to fix the problem: "the dropdown is not editable but the screen reader keeps saying it is".

That is fixed using aria-readonly in this PR.

**Comment by micahgodbolt** [comment]
I know @betrue-final-final was at an all hands yesterday. hopefully he's had 1 more chance to look at this

**Comment by betrue-final-final** [comment]
I'll check this out.

**Comment by betrue-final-final** [comment]
Checking this out now.

**Comment by betrue-final-final** [comment]
Oh, right. Yeah, if this is called "read-only" I would imagine it would create confusion as to if it actually works to change a value. I also found a couple of accessibility bugs unrelated to the PR.

**Comment by jspurlin** [comment]
This should not be using aria-readonly, that means that the value cannot be changed which is not correct. Instead this should use aria-autocomplete="none"

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -158,7 +158,8 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
           id={ id }
           tabIndex={ disabled ? -1 : 0 }
           aria-expanded={ isOpen ? 'true' : 'false' }
-          role='menu'
+          role='combobox'
+          aria-readonly='true'
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: this should not be aria-readonly="true" instead it should be aria-autocomplete="none"

**Comment by cschlechty** [comment]
Does using role=listbox make more sense? comboboxes are always textbox+menu

**Comment by cschlechty** [comment]
looks good

**Pull request merged**